### ISSUE TITLE
Table (edit mode) - not accessible with screen readers

### ISSUE THREAD

**Issue by CarmenHilpert** [ckeditor/ckeditor5#19157]
## 📝 Provide detailed reproduction steps (if any)

1. Open https://ckeditor.com/docs/ckeditor5/latest/features/tables/tables.html
2. Place cursor above the table
4. Press the arrow down key (the horizontal cursor becomes visible)
5. Press the arrow down key (the table receives focus)
6. Press the Tab key (the first cell receives focus)

### ✔️ Expected result

- The screen reader user should receive clear infromation about the current position and available actions.

- The screen reader should announce:
   - "insert a line before table <caption text>"
   - "Table with 4 columns and 6 rows"
   - The caption: "Countries that sent their citizen into space"
   - Braille: r1c1 "Country"

The help (Alt+0) should provide a hint that tables can be entered using the"Tab" key.

### ❌ Actual result

- Screen reader output is minimal or unclear:
   - "t"
   - (silence, no output)
   - Braille: r1c1 "Country"

Even when reading the help (Alt+0), the users obtains no guidance on how to enter a table. 

## 📃 Other details

* Browser: Edge 138.0.3351.65
* OS: Windows 11 (10.0.22631 Build 22631)
* Screenrader: JAWS 2024
* First affected CKEditor version: not shure when it started - seen in v44.3 and later
* Installed CKEditor plugins: see demo page

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.
